models/patch.go

```diff
@@ -76,12 +76,12 @@ func (db *DB) GetPatches(filter *Filter) ([]Patch, error) {
 	queryString := new(strings.Builder)
 	fmt.Fprintf(queryString, "SELECT * FROM patches WHERE %s", filterString)
 	rows, err := db.Query(queryString.String())
-	defer rows.Close()
 	if err != nil {
 		log.Print("Error getting rows")
 		log.Print(err)
 		return nil, err
 	}
+	defer rows.Close()
```

Comment:
Was this the segfault?

Reply:
Yeah.

Reply:
`rows` was `nil` when there was an error, so the `defer` would cause a segault.

Reply:
Amazing, that's what I thought. This was the main thing I was looking for in the PR lol.